### DISCUSSION TITLE
Should `"diagnostic.separateRelatedInformationAsDiagnostics"` feature be implemented by coc client rather than LS?

### DISCUSSION THREAD

**Discussion by hexh250786313** [neoclide/coc.nvim#5078] [General]
`"Related diagnostics:"` message not showed in coc-tsserver because coc-tsserver doesn't implement the `"diagnostic.separateRelatedInformationAsDiagnostics"` feature. I think this feature should be implemented in the public `diagnostic/collection.ts` rather than LS itself.

Any suggestions?

**Comment by fannheyward** [comment]
> because coc-tsserver doesn't implement

does tsserver returns `diagnostic.relatedInformation`?

**Reply by hexh250786313** [reply]
Yes, it returns.

Coc.nvim implements this feature for all LS configured in `"languageserver"`. But for standalone LS extension, it depends on whether the extension itself implements this feature or not.

https://github.com/neoclide/coc.nvim/blob/a00ff123abe37a99962fd4c64b83479a9b68bbde/src/language-client/client.ts#L1580-L1596
